build.gradle

```diff
@@ -244,6 +244,12 @@ subprojects {
       }
     }
 
+    slf4jApi = { ->
+     dependencies {
+        implementation group: 'org.slf4j', name: 'slf4j-api', version: slf4jVersion
```

Comment:
1) rename `slf4j` to `slf4jTestImpl`
2) see the comment above for `slf4j`, in non-test contexts we're basically bound to use whatever `slf4j-api` version IntelliJ uses, currently it's 1.7.28... and `slf4jVersion = 1.7.30` (the current latest).

So either change `slf4jVersion` to `1.7.28` to make it equal to the IntelliJ's `slf4j-api` version, or preferably split it into `slf4jApiVersion` (which must be 1.7.28 or we might have a trouble at some point in the runtime) and `slf4jTestImplVersion` (which can be basically anything, it's only used for tests anyway)

Reply:
3. `slf4jTestImpl` should likely exclude transitive dep to `slf4j-api` so that we never end up with two different `slf4j-api` versions in test configs (one being `slf4jApi` that the tested code depends on, and one being transitively pulled via `slf4jTestImpl`)

Reply:
The general idea behind all that - in every possible config where stuff can be logged (tests, `buildPlugin`/`runIde`), we need to provide `slf4j-api` in exactly one version and exactly one slf4j impl (not even two copies of the same impl in the same version AFAIK!) on the classpath.

`buildPlugin`/`runIde` configs are special coz they provide both already - so we need to make sure that we only provide `slf4j-api` in a consistent version, and **no SLF4J** impl at all.